File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_4_server_monitor/after/tests/FakeAlertPort.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.tests;

import com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.core.ports.AlertPort;
import java.util.ArrayList;
import java.util.List;

public class FakeAlertPort implements AlertPort {
    private final List<String> sentMessages = new ArrayList<>();

    @Override
    public void sendAlert(String message) {
        sentMessages.add(message);
    }

    public List<String> getSentMessages() {
        return sentMessages;
    }
}

File: Chapter 05/C#/Section_5_4_Server_Monitor/After/Tests/FakeAlertPort.cs
using System.Collections.Generic;
using Chapter05.ServerMonitor.After.Core.Ports;

namespace Chapter05.ServerMonitor.After.Tests
{
    public class FakeAlertPort : IAlertPort
    {
        public List<string> SentMessages { get; } = new List<string>();
        public void SendAlert(string message) => SentMessages.Add(message);
    }
}
